core/src/main/scala/com/salesforce/op/filters/FeatureDistribution.scala

```diff
@@ -225,17 +246,53 @@ object FeatureDistribution {
       value.map(seq => 0L -> histValues(seq, summary, bins, textBinsFormula))
         .getOrElse(1L -> (Array(summary.min, summary.max, summary.sum, summary.count) -> new Array[Double](bins)))
 
+    val momentsAndCard = value match {
+      case Some(m) => (Some(momentsValues(m)), Some(cardinalityValues(m)))
+      case _ => (None, None)
+    }
+
     FeatureDistribution(
       name = name,
       key = key,
       count = 1L,
       nulls = nullCount,
       summaryInfo = summaryInfo,
       distribution = distribution,
+      moments = momentsAndCard._1,
+      cardEstimate = momentsAndCard._2,
```

Comment:
why put it in a tuple and then untuple?
maybe just 
`moments = m.map(momentsValues)`
`cardEstimate = m.map(cardinalityValues)`

Reply:
moments = value.map(momentsValues),
      cardEstimate = value.map(cardinalityValues),

Reply:
delete momentsAndCard